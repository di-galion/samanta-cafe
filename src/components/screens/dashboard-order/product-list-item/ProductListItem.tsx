import styles from "./styles.module.scss"
import {FC} from "react";
import {IProduct} from "@/types/product.types";
import Image from "next/image";
import {ISailCreate} from "@/types/sail.types";

const ProductListItem: FC<{product: IProduct, setNewSailOnClick: (sail: any) => void, orderId: number}> = (
    {
        product,
        setNewSailOnClick,
        orderId
    }) => {
    return (
        <li
            data-testid={"product-list-item-test"}
            onClick={() => setNewSailOnClick(
                {
                    name: product.name,
                    price: product.price,
                    quantity: 1,
                    productId: product.id,
                    orderId: orderId
                })}
            className={styles.product_item}
        >
            <Image
                src={product.images[0]}
                height={50}
                width={50}
                alt={"picture"}
            />
            <div className={styles.info_container}>
                <div className={styles.text}>
                    {product.name}
                </div>
                <div className={styles.description}>
                    {product.description}
                </div>
            </div>
        </li>
    )
}

export default ProductListItem