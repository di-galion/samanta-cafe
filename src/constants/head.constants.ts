export const ORDER_ARRAY_OF_HEAD = ["Имя", "Телефон", "Улица", "Дом", "Подьезд", "Этаж","Комната", "Ключ", "Комментарий"]

export const PRODUCT_ARRAY_OF_HEAD = ["Название", "Описание", "Цена"]

export const CATEGORY_ARRAY_OF_HEAD = ["Название", "Slug"]