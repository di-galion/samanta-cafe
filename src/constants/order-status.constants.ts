export const NO_FILTER = 'No filtration'