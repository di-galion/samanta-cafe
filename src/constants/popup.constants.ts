export const POPUP_ID = "popup"
export const POPUP_ITEM_TEXT  = "Do you want to save edit?"