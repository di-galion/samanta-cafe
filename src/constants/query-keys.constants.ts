export const GET_ALL_ORDERS = "get all orders"
export const UPDATE_ORDER = "update order"
export const CREATE_ORDER = "create order"


export const CREATE_SAIL = "create sail"
export const UPDATE_SAIL = "update sail"

export const DELETE_SAIL = "delete sail"

export const CREATE_CATEGORY = "create category"
export const UPDATE_CATEGORY = "update category"
export const GET_ALL_CATEGORIES = 'get all categories'

export const GET_ALL_PRODUCTS = "get all products"
export const CREATE_PRODUCT = "create product"
export const UPDATE_PRODUCT = "update product"

export const GET_PROFILE = 'get profile'