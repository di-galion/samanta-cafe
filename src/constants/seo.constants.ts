export const SITE_NAME = "Samanta"