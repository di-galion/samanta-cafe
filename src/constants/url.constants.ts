export const ADMIN_URL = '/admin'
export const ORDER_URL = '/orders'
export const CATEGORY_URL = "/category"
export const PRODUCT_URL = "/product"

export const AUTH_URL = '/auth'

export const USER_URL = "/user"

export const DASHBOARD_URL = "/dashboard"