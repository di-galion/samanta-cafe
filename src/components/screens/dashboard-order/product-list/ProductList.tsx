"use client"
import styles from "./styles.module.scss"
import {useQuery} from "@tanstack/react-query";
import {OrderService} from "@/services/order/order.service";
import {ProductService} from "@/services/product/product.service";
import ProductListItem from "@/components/screens/dashboard-order/product-list-item/ProductListItem";
import {IProduct} from "@/types/product.types";
import {FC} from "react";
import {ISailCreate} from "@/types/sail.types";
import Loader from "@/components/ui/loader/Loader";
import {GET_ALL_PRODUCTS} from "@/constants/query-keys.constants";
import {useGetProducts} from "@/components/screens/dashboard-order/product-list/useGetProducts.hook";

export interface IProductList {
    setNewSailOnClick: (sail: ISailCreate) => void
    orderId: number
}
const ProductList: FC<IProductList> = ({setNewSailOnClick, orderId}) => {

    const {data, isLoading} = useGetProducts()

    return (
        <div className={styles.product_list_overlay}>
            <ul className={styles.product_list}>
                {!isLoading && data ?
                        data.map((product: IProduct)=> {
                            return (
                                <ProductListItem
                                    key={product.id}
                                    setNewSailOnClick={setNewSailOnClick}
                                    product={product}
                                    orderId={orderId}
                                />
                            )
                        }) :
                    <Loader />
                }
            </ul>
        </div>
    )
}

export default ProductList