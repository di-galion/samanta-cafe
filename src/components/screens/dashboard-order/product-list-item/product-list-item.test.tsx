import {useGetProducts} from "@/components/screens/dashboard-order/product-list/useGetProducts.hook";
import {PRODUCTS_TEST} from "@/components/screens/product/__tests__/constants";
import {renderWithProviders} from "@/__tests__/helpers/renderTestApp";
import ProductList from "@/components/screens/dashboard-order/product-list/ProductList";
import {fireEvent, screen} from "@testing-library/react";
import ProductListItem from "@/components/screens/dashboard-order/product-list-item/ProductListItem";

describe("Product list item", () => {

    afterEach(() => {
        jest.clearAllMocks()
    })

    test("Product list item render", () => {

        const component = renderWithProviders(<ProductListItem product={PRODUCTS_TEST[0]} setNewSailOnClick={jest.fn()} orderId={1} />)

        expect(component.getByRole("img"))
        expect(component.getByText(PRODUCTS_TEST[0].description))
        expect(component.getByText(PRODUCTS_TEST[0].name))
        expect(component).toMatchSnapshot()
    })

    test("Should call setNewSailOnClick with click on 'product-list-item-test'", () => {
        const setNewSailOnClick = jest.fn()
        const component = renderWithProviders(<ProductListItem product={PRODUCTS_TEST[0]} setNewSailOnClick={setNewSailOnClick} orderId={1} />)

        fireEvent.click(component.getByTestId("product-list-item-test"))

        expect(setNewSailOnClick).toHaveBeenCalledTimes(1)
    })

})