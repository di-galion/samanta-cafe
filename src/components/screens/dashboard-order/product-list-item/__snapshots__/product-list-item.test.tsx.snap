// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list item Product list item render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="product_item"
        data-testid="product-list-item-test"
      >
        <img
          alt="picture"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Flocal&w=128&q=75"
          srcset="/_next/image?url=http%3A%2F%2Flocal&w=64&q=75 1x, /_next/image?url=http%3A%2F%2Flocal&w=128&q=75 2x"
          style="color: transparent;"
          width="50"
        />
        <div
          class="info_container"
        >
          <div
            class="text"
          >
            f
          </div>
          <div
            class="description"
          >
            Some description
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="product_item"
      data-testid="product-list-item-test"
    >
      <img
        alt="picture"
        data-nimg="1"
        decoding="async"
        height="50"
        loading="lazy"
        src="/_next/image?url=http%3A%2F%2Flocal&w=128&q=75"
        srcset="/_next/image?url=http%3A%2F%2Flocal&w=64&q=75 1x, /_next/image?url=http%3A%2F%2Flocal&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
      <div
        class="info_container"
      >
        <div
          class="text"
        >
          f
        </div>
        <div
          class="description"
        >
          Some description
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
